pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -215,9 +216,51 @@ func getRoleMetaObject(role interface{}, roleType RoleType) (meta *metav1.Object
 	return
 }
 
-func areRoleRulesEqual(role1 interface{}, role2 interface{}, roleType RoleType) (equal bool) {
+func enforceAPIGroup(existing interface{}, required interface{}, roleType RoleType) {
+	var existingRoleRef *rbacv1.RoleRef
+	var requiredRoleRef *rbacv1.RoleRef
+	var existingSubjects []rbacv1.Subject
+	var requiredSubjects []rbacv1.Subject
+
+	switch roleType {
+	case TypeRole:
+		fallthrough
+	case TypeClusterRole:
+		return
+	case TypeRoleBinding:
+		crExisting := existing.(*rbacv1.RoleBinding)
+		crRequired := required.(*rbacv1.RoleBinding)
+		existingRoleRef = &crExisting.RoleRef
+		requiredRoleRef = &crRequired.RoleRef
+		existingSubjects = crExisting.Subjects
+		requiredSubjects = crRequired.Subjects
+	case TypeClusterRoleBinding:
+		crbExisting := existing.(*rbacv1.ClusterRoleBinding)
+		crbRequired := required.(*rbacv1.ClusterRoleBinding)
+		existingRoleRef = &crbExisting.RoleRef
+		requiredRoleRef = &crbRequired.RoleRef
+		existingSubjects = crbExisting.Subjects
+		requiredSubjects = crbRequired.Subjects
```

Comment:
I know statically any other type than the ones you listed will not be passed to this function but since the params of this are generic someone might be tempted to pass something that you don't expect leading to undefined behaviour, advise you can freely ignore: I like adding a 

````
default:
    logThatThisTypeShouldNotHaveBeenPassedHere()
    return
```
to functions like this one, just for safety.

Reply:
Thanks Antonio.

The thing is that this function is called also for types are do not need to enforce API group. As part of the refactoring there is one generic function in rbac to deal with any one of the resources. So actually it is expected that "unsupported" types will be passed to this function and then nothing will be done (i.e. the function will return). Since we're talking about a private helper function and can be called only within rbac I think that this behavior can be considered safe.

Reply:
Yes, and I saw you return immediately if you see you have `TypeRole` or a `TypeClusterRole`, was just suggesting that you could delete those 2 cases and just add a default as a catch-all (in case something else entirely gets passed that is not between those 4 types), but as I said feel free to not do it as I'm probably overthinking it :)

Reply:
Oh now I see your point :)
Yes, I would totally agree in almost any other context, but since this is in the specific rbac context I'm not sure it's worth adding those lines of code to every one of the helper functions. Also, you have to pass a `RoleType` parameter which implies that not every type is supported or expected. And also - there are unit tests that cover those cases :)